library/src/webapp/js/spinner.js

```diff
@@ -23,8 +23,11 @@
 // Spinner namespace
 var SPNR = SPNR || {};
 
-// Get the main content div
-const mainContent = () => document.getElementById("content");
+// Get the main content div, return an empty div if it can't be found
+const mainContent = () => {
+  const contentElement = document.getElementById("content");
+  return contentElement !== null ? contentElement : document.createElement("div");
```

Comment:
Should we also need to append the created `div` element to the dom?

Reply:
You are right, it's smart to do so. div doesn't need to be added to the DOM since it serves as a safety net nothing more.